### PR TITLE
Send addConnection telemetry on builder id login

### PR DESCRIPTION
**Pull request by wularr**
## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description

- Send addConnection telemetry on builder id login
- Send loginWithBrowser failure only if builder id login failed

## Checklist
- [X] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [X] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by rli — APPROVED** [review]

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt [hunk 1]
```diff
@@ -96,14 +97,21 @@ abstract class LoginBrowser(
...
+        val onSuccess: () -> Unit = {
             AwsTelemetry.loginWithBrowser(
                 project = null,
                 credentialStartUrl = SONO_URL,
                 result = Result.Succeeded,
                 credentialSourceId = CredentialSourceId.AwsId
             )
+            AuthTelemetry.addConnection(
+                result = Result.Succeeded,
+                credentialSourceId = CredentialSourceId.AwsId
```
Comment (manodnyab) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt:109]: Should we also add this metric for on onError? @dhruvigajjar

**Pull request merged**